governance/xc_admin/packages/xc_admin_common/src/programs/types.ts

```diff
@@ -0,0 +1,254 @@
+import {
+  AccountInfo,
+  PublicKey,
+  TransactionInstruction,
+} from "@solana/web3.js";
+import { PermissionData, Product, PythCluster } from "@pythnetwork/client";
+import { Connection } from "@solana/web3.js";
+import { Program } from "@coral-xyz/anchor";
+import { PythOracle } from "@pythnetwork/client/lib/anchor";
+import { MessageBuffer } from "message_buffer/idl/message_buffer";
+/**
+ * Represents the different Pyth programs supported by the application.
+ */
+export enum ProgramType {
+  /**
+   * Original Pyth oracle program
+   */
+  PYTH_CORE,
+
+  /**
+   * Next-generation Pyth oracle program
+   */
+  PYTH_LAZER,
+}
+
+/**
+ * Human-readable names for program types
+ */
+export const PROGRAM_TYPE_NAMES: Record<ProgramType, string> = {
+  [ProgramType.PYTH_CORE]: "Pyth Core",
+  [ProgramType.PYTH_LAZER]: "Pyth Lazer",
+};
+
+/**
+ * Type for raw price configs
+ */
+export type PriceRawConfig = {
+  next: PublicKey | null;
+  address: PublicKey;
+  expo: number;
+  minPub: number;
+  maxLatency: number;
+  publishers: PublicKey[];
+};
+
+/**
+ * Type for raw product configs
+ */
+export type ProductRawConfig = {
+  address: PublicKey;
+  priceAccounts: PriceRawConfig[];
+  metadata: Product;
+};
+
+/**
+ * Type for raw mapping configs
+ */
+export type MappingRawConfig = {
+  address: PublicKey;
+  next: PublicKey | null;
+  products: ProductRawConfig[];
+};
+
+/**
+ * Overall raw configuration type
+ */
+export type RawConfig = {
+  mappingAccounts: MappingRawConfig[];
+  permissionAccount?: PermissionData;
+};
+
+export type CoreConfigParams = {
+  accounts: Array<{ pubkey: PublicKey; account: AccountInfo<Buffer> }>;
+  cluster: PythCluster;
+};
+
+/**
+ * Lazer-specific configuration type
+ * TODO: Change to actual Lazer config type
+ */
+export type LazerConfig = {
+  programType: ProgramType.PYTH_LAZER;
+  // Make cluster optional since Lazer might not be tied to a specific cluster
+  cluster?: PythCluster;
+  // More generic data source instead of Solana-specific accounts
+  feeds: LazerFeed[];
+  // Additional metadata that might be relevant for Lazer
+  metadata?: Record<string, unknown>;
+};
+
+/**
+ * Parameters for getting Lazer configuration
+ */
+export type LazerConfigParams = {
+  // Instead of requiring Solana accounts, allow any parameters needed
+  endpoint?: string;
+  network?: string;
+  options?: Record<string, unknown>;
+};
+
+/**
+ * Union type for configuration parameters that can vary by program type
+ */
+export type GetConfigParams =
+  | ({
+      programType: ProgramType.PYTH_CORE;
+    } & CoreConfigParams)
+  | ({ programType: ProgramType.PYTH_LAZER } & LazerConfigParams);
+
+/**
+ * Function to get configuration data
+ */
+export type GetConfigFn = (params: GetConfigParams) => ProgramConfig;
```

Comment:
Types like this should not be necessary; in general it's best to let typescript infer function types.  If you must hint the function type (e.g. if the body is complex and you want to guarantee you're returning what you expect), the you can do that by adding the return type to the function definition rather than letting typescript infer the return type.